### PR TITLE
itest+lntest: speed up test setup

### PR DESCRIPTION
**Pull request by yyforyongyu**
Turns out we are doing 200 `GetRawMempool` RPC calls when setting up the test case, which is now optimized.

Before this change,
```
--- PASS: TestLightningNetworkDaemon (68.79s)
    --- PASS: TestLightningNetworkDaemon/tranche00/17-of-174/btcd/list_channels (13.86s)
```

After this change,
```
--- PASS: TestLightningNetworkDaemon (29.66s)
    --- PASS: TestLightningNetworkDaemon/tranche00/17-of-174/btcd/list_channels (13.89s)
```

Also changes the methods so we don't pass pointers of `chainhash.Hash` anymore in the tests.

Thanks @bitromortac for pointing this out!

### CODE REVIEW

File: lntest/harness.go [hunk 1]
```diff
@@ -342,6 +342,7 @@ func (h *HarnessTest) SetupStandbyNodes() {
 	// above a good number of confirmations.
 	const totalTxes = 200
 	h.MineBlocksAndAssertNumTxes(numBlocksSendOutput, totalTxes)
+	h.MineBlocks(numBlocksSendOutput)
```
Comment (bitromortac) [inline lntest/harness.go:345]: nit: may move this change to its own commit, the motivation is not clear

File: lntest/miner/miner.go [hunk 2]
```diff
@@ -314,6 +314,34 @@ func (h *HarnessMiner) AssertTxInMempool(txid chainhash.Hash) *wire.MsgTx {
 	return h.GetRawTransaction(txid).MsgTx()
 }
 
+// AssertTxnsNotInMempool asserts the given txns are not found in the mempool.
+// It assumes the mempool is not empty.
+func (h *HarnessMiner) AssertTxnsNotInMempool(txids []chainhash.Hash) {
+	err := wait.NoError(func() error {
+		// We require the RPC call to be succeeded and won't wait for
+		// it as it's an unexpected behavior.
```
Comment (bitromortac) [inline lntest/miner/miner.go:322]: nit: not sure I understand that comment

**Review by bhandras — APPROVED** [review]
Nice! LGTM 🎉

**Review by bitromortac — APPROVED** [review]
LGTM :tada:, great improvement for single itest runs, thanks for the investigation!

File: lntest/miner/miner.go (hunk 2 above)
Comment (guggero) [reply lntest/miner/miner.go:322]: Seems to be copy/paste, appears in a couple of other places. I interpret it as: "this will call `ht.t.Fail()` if it doesn't succeed, which will abort the test, so we can continue below assuming it didn't fail".

**Review by guggero — APPROVED** [review]
Very nice!
